Change to use pyomnisci instead of pymapd

There's new functionality in pyomnisci, so it would be good to use that instead of pymapd.

https://github.com/omnisci/pyomnisci

https://pypi.org/project/pyomnisci/

It seems also it will need a new release for pyomniscidb as the current release is pinning rbc to 0.2.2 

https://github.com/omnisci/omniscidb/blob/v5.5.2/python/setup.py

and this version is using pymapd:

/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/rbc/omniscidb.py:8: in <module>
    from pymapd.cursor import make_row_results_set
E   ModuleNotFoundError: No module named 'pymapd'

any plans for a new release for `pyomniscidb`?

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -44,8 +44,7 @@ jobs:
     - name: install dependencies
       run: |
         if [[ "${{ matrix.pyomnisci-version }}" == "dev" ]]; then
-          pip install git+https://github.com/omnisci/pyomnisci.git
-          pip install rbc-project>=0.4.0
+          pip install --no-deps git+https://github.com/omnisci/pyomnisci.git
         fi
         python -m pip install --no-deps -e .
 

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -44,7 +44,7 @@ jobs:
     - name: install dependencies
       run: |
         if [[ "${{ matrix.pymapd-version }}" == "dev" ]]; then
-          pip install git+https://github.com/omnisci/pyomnisci.git
+          pip install git+https://github.com/omnisci/pyomnisci.git rbc-project>=0.4.0
         fi
         python -m pip install --no-deps -e .
 

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -17,7 +17,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        pymapd-version: ["stable", "dev"]
+        pyomnisci-version: ["stable", "dev"]
 
     defaults:
       run:
@@ -43,8 +43,9 @@ jobs:
 
     - name: install dependencies
       run: |
-        if [[ "${{ matrix.pymapd-version }}" == "dev" ]]; then
-          pip install git+https://github.com/omnisci/pyomnisci.git rbc-project>=0.4.0
+        if [[ "${{ matrix.pyomnisci-version }}" == "dev" ]]; then
+          pip install git+https://github.com/omnisci/pyomnisci.git
+          pip install rbc-project>=0.4.0
         fi
         python -m pip install --no-deps -e .
 

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -44,7 +44,7 @@ jobs:
     - name: install dependencies
       run: |
         if [[ "${{ matrix.pymapd-version }}" == "dev" ]]; then
-          pip install git+https://github.com/omnisci/pymapd.git
+          pip install git+https://github.com/omnisci/pyomnisci.git
         fi
         python -m pip install --no-deps -e .
 

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -1,6 +1,6 @@
 repos:
   - repo: https://github.com/PyCQA/isort
-    rev: main  # stable
+    rev: 5.8.0
     hooks:
       - id: isort
 

--- a/ci/setup_tests.py
+++ b/ci/setup_tests.py
@@ -12,7 +12,7 @@ import urllib.request
 import zipfile
 
 import pandas
-import pymapd
+import pyomnisci
 
 BASE_PATH = os.path.abspath(os.path.dirname(__file__))
 
@@ -74,7 +74,7 @@ def main(repo_url, schema, tables, data_directory, **params):
     database = params["database"]
 
     if database != default_db:
-        conn = pymapd.connect(
+        conn = pyomnisci.connect(
             host=params['host'],
             user=params['user'],
             password=params['password'],
@@ -95,7 +95,7 @@ def main(repo_url, schema, tables, data_directory, **params):
             print(f'OmniSci DDL statement {stmt} failed', file=sys.stderr)
         conn.close()
 
-    conn = pymapd.connect(
+    conn = pyomnisci.connect(
         host=params['host'],
         user=params['user'],
         password=params['password'],

--- a/environment-dev.yaml
+++ b/environment-dev.yaml
@@ -6,11 +6,19 @@ dependencies:
 
 # core
 - ibis-framework  # TODO: require Ibis 2.0 when it's released
-- pymapd>=0.24
-- pyarrow>=0.15.0
+- pandas
+- pyomnisci>=0.27.0
+- pyomniscidb>=5.5.2
+- pyarrow
+# numpy 1.20 has conflicts with pandas and pyarrow
+# pyarrow.lib.ArrowTypeError: ('Did not pass numpy.dtype object',
+#   'Conversion failed for column salary with type float64')
 - numpy<1.20
-- rbc
+- rbc>=0.4.0
+# ibis 1.4 for python 3.7 installs pymapd, and ibis-omniscidb
+# doens't use pymapd anymore
 - python 3.8.*
+
 # TODO: sqlalchemy pinning should be removed when Ibis 2.0 is released
 - sqlalchemy<1.4
 

--- a/ibis_omniscidb/client.py
+++ b/ibis_omniscidb/client.py
@@ -8,16 +8,15 @@ import ibis.expr.operations as ops
 import ibis.expr.schema as sch
 import ibis.expr.types as ir
 import pandas as pd
-import pkg_resources
 import pyarrow
-import pymapd
+import pyomnisci
 import regex as re
 from ibis.backends.base_sqlalchemy.compiler import DDL, DML
 from ibis.client import Database, DatabaseEntity, Query, SQLClient
 from ibis.util import log
-from pymapd._parsers import _extract_column_details
-from pymapd.cursor import Cursor
-from pymapd.dtypes import TDatumType as pymapd_dtype
+from omnisci._parsers import _extract_column_details
+from omnisci.cursor import Cursor
+from omnisci.dtypes import TDatumType as pyomnisci_dtype
 
 from . import ddl
 from . import dtypes as omniscidb_dtypes
@@ -55,12 +54,6 @@ def _validate_compatible(from_schema, to_schema):
                 'Cannot safely cast {0!r} to {1!r}'.format(lt, rt)
             )
     return
-
-
-class PyMapDVersionError(Exception):
-    """PyMapD version error exception."""
-
-    pass
 
 
 class OmniSciDBDataType:
@@ -203,14 +196,14 @@ class OmniSciDBGeoCursor(OmniSciDBDefaultCursor):
         col_names = [c.name for c in cursor_description]
         result = pd.DataFrame(cursor.fetchall(), columns=col_names)
 
-        # get geo types from pymapd
+        # get geo types from pyomnisci
         geotypes = (
-            pymapd_dtype.POINT,
-            pymapd_dtype.LINESTRING,
-            pymapd_dtype.POLYGON,
-            pymapd_dtype.MULTIPOLYGON,
-            pymapd_dtype.GEOMETRY,
-            pymapd_dtype.GEOGRAPHY,
+            pyomnisci_dtype.POINT,
+            pyomnisci_dtype.LINESTRING,
+            pyomnisci_dtype.POLYGON,
+            pyomnisci_dtype.MULTIPOLYGON,
+            pyomnisci_dtype.GEOMETRY,
+            pyomnisci_dtype.GEOGRAPHY,
         )
 
         geo_column = None
@@ -611,8 +604,6 @@ class OmniSciDBClient(SQLClient):
         ------
         Exception
             if the given execution_type is not valid.
-        PyMapDVersionError
-            if session_id is given but pymapd version is less or equal to 0.12
         """
         self.uri = uri
         self.user = user
@@ -629,11 +620,7 @@ class OmniSciDBClient(SQLClient):
         self.gpu_device = gpu_device
 
         if session_id:
-            if self.version < pkg_resources.parse_version('0.12.0'):
-                raise PyMapDVersionError(
-                    'Must have pymapd > 0.12 to use session ID'
-                )
-            self.con = pymapd.connect(
+            self.con = pyomnisci.connect(
                 uri=uri,
                 host=host,
                 port=port,
@@ -641,7 +628,7 @@ class OmniSciDBClient(SQLClient):
                 sessionid=session_id,
             )
         else:
-            self.con = pymapd.connect(
+            self.con = pyomnisci.connect(
                 uri=uri,
                 user=user,
                 password=password,
@@ -748,7 +735,7 @@ class OmniSciDBClient(SQLClient):
             (
                 r.col_name,
                 OmniSciDBDataType._omniscidb_to_ibis_dtypes[
-                    pymapd_dtype._VALUES_TO_NAMES[r.col_type.type]
+                    pyomnisci_dtype._VALUES_TO_NAMES[r.col_type.type]
                 ],
             )
             for r in result
@@ -1235,7 +1222,7 @@ class OmniSciDBClient(SQLClient):
         """
         if self.db_name != name and name is not None:
             self.con.close()
-            self.con = pymapd.connect(
+            self.con = pyomnisci.connect(
                 uri=self.uri,
                 user=self.user,
                 password=self.password,
@@ -1375,9 +1362,7 @@ class OmniSciDBClient(SQLClient):
         string
             Version of the backend library.
         """
-        # pymapd doesn't have __version__
-        dist = pkg_resources.get_distribution('pymapd')
-        return pkg_resources.parse_version(dist.version)
+        return pyomnisci.__version__
 
 
 @dt.dtype.register(OmniSciDBDataType)

--- a/ibis_omniscidb/tests/conftest.py
+++ b/ibis_omniscidb/tests/conftest.py
@@ -9,6 +9,9 @@ import pytest
 
 import ibis_omniscidb
 
+# NOTE: TEMPORARY UNTIL IBIS 2.0 IS RELEASED
+ibis.omniscidb = ibis_omniscidb
+
 OMNISCIDB_HOST = os.environ.get('IBIS_TEST_OMNISCIDB_HOST', 'localhost')
 OMNISCIDB_PORT = int(os.environ.get('IBIS_TEST_OMNISCIDB_PORT', 6274))
 OMNISCIDB_USER = os.environ.get('IBIS_TEST_OMNISCIDB_USER', 'admin')

--- a/ibis_omniscidb/tests/test_client.py
+++ b/ibis_omniscidb/tests/test_client.py
@@ -7,11 +7,15 @@ import ibis.expr.types as ir
 import mock
 import pandas as pd
 import pyarrow
-import pymapd
+import pyomnisci
 import pytest
 from ibis.tests.util import assert_equal
-from pkg_resources import get_distribution, parse_version
 from pytest import param
+
+import ibis_omniscidb
+
+# NOTE: TEMPORARY UNTIL IBIS 2.0 IS RELEASED
+ibis.omniscidb = ibis_omniscidb
 
 
 def test_table(alltypes):
@@ -42,10 +46,6 @@ def test_list_tables(con):
     assert len(con.list_tables(like='functional_alltypes')) == 1
 
 
-@pytest.mark.skipif(
-    parse_version(get_distribution('pymapd').version) < parse_version('0.12'),
-    reason='must have pymapd>=12 to connect to existing session',
-)
 def test_session_id_connection(session_con):
     new_connection = ibis.omniscidb.connect(
         protocol=session_con.protocol,
@@ -268,7 +268,7 @@ def test_cpu_execution_type(
 
     for mock_method_name in ('select_ipc', 'select_ipc_gpu'):
         mocked_method = mock.patch.object(
-            pymapd.connection.Connection,
+            pyomnisci.connection.Connection,
             mock_method_name,
             new=lambda *args, **kwargs: pd.DataFrame({'string_col': ['1']}),
         )

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,9 +1,17 @@
 # core
 ibis-framework
-pymapd>=0.24
-pyarrow>=0.15.0
+pandas
+pyomnisci>=0.27.0
+pyomniscidb>=5.5.2
+pyarrow
+# numpy 1.20 has conflicts with pandas and pyarrow
+# pyarrow.lib.ArrowTypeError: ('Did not pass numpy.dtype object',
+#   'Conversion failed for column salary with type float64')
 numpy<1.20
-rbc-project
+rbc-project>=0.4.0
+
+# temporary fix
+sqlalchemy<1.4
 
 # dev
 black

--- a/setup.py
+++ b/setup.py
@@ -14,9 +14,17 @@ setuptools.setup(
     packages=setuptools.find_packages(),
     python_requires='>=3.7',
     install_requires=[
-        'ibis-framework',  # TODO require ibis 2.0 when it's released
-        'pymapd>=0.24',
+        'ibis-framework',  # TODO: require ibis 2.0 when it's released
+        'pandas',
+        'pyomnisci>=0.27.0',
+        'pyomniscidb>=5.5.2',
         'pyarrow',
+        'rbc-project>=0.4.0',
+        'sqlalchemy<1.4',  # TODO: it should be fixed by ibis 2.0
+        # numpy 1.20 has conflicts with pandas and pyarrow
+        # pyarrow.lib.ArrowTypeError: ('Did not pass numpy.dtype object',
+        #   'Conversion failed for column salary with type float64')
+        'numpy<1.20',
     ],
     setup_requires=['setuptools_scm'],
     use_scm_version=True,

